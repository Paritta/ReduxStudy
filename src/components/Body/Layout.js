import React from "react";
import styled from "styled-components";
import Card from "./Card";
import PropTypes from "prop-types";
import ReactLoading from "react-loading";

const propTypes = {
    Fetch: PropTypes.object,
    Filter: PropTypes.object,
    showModal: PropTypes.func,
    hideModal: PropTypes.func,
    fetchRequest: PropTypes.func,
    imageReceiveRequest: PropTypes.func,
    ImageReceive: PropTypes.object,
};

const defaultTypes = {
    Fetch: {},
    Filter: {},
    showModal() {},
    hideModal() {},
    fetchRequest() {},
    imageReceiveRequest() {},
    ImageReceive: {},
};

const Wrapper = styled.div`
    position: relative;
    
    width: 100%;
    height: 100vh;
    
    padding: 0 40px;
    text-align: left;
`;

const WrappedReactLoading = styled.div`
    transform: scale(2);
    position: absolute;
    
    top: 50%;
    left: 50%;
    
    padding: 80px 0;
`;

const Item = styled.div`
    display: inline-block;
    width: 33.3%;
    margin-bottom: 20px;
    margin-top: 60px;
`;

export class Layout extends React.Component {
    componentDidMount () {
        this.props.fetchRequest();
    }

    render () {
        const { Fetch, showModal, imageReceiveRequest, ImageReceive } = this.props;
        const Pending = Fetch.pending;

        return (
            <div>
                <Wrapper>
                    {
                        Pending?
                        <WrappedReactLoading>
                            <ReactLoading type="cylon" color="palevioletred"/>
                        </WrappedReactLoading>
                        :
                        Fetch.data.map((item, key) =>
                            <Item
                                onClick={() =>
                                    { showModal({
                                        modalType: "MODAL_STUDY",
                                        modalProps: item
                                    })}}
                                key={key}
                            >
                                <Card
                                    item={item}
                                    postId={item.postId}
                                    imageReceiveRequest={imageReceiveRequest}
                                    ImageReceive={ImageReceive}
                                />
                            </Item>
                        )
                    }
                </Wrapper>
            </div>
        );
    }
}

Layout.propTypes = propTypes;
Layout.defaultTypes = defaultTypes;

export default Layout;
