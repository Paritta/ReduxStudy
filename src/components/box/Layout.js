import React from "react";
import styled from "styled-components";
import Card from "./Card";
import PropTypes from "prop-types";
import ReactLoading from "react-loading";

const propTypes = {
    showModal: PropTypes.func,
};

const defaultTypes = {
    showModal () {},
};

const Wrapper = styled.div`
    position: relative;
    width: 100%;
    padding: 100px;
`;

const WrappedReactLoading = styled.div`
    transform: scale(2);
    position: absolute;
    
    top: 50%;
    left: 50%;
    
    padding: 80px 0;
`;

const Item = styled.div`
    display: inline-block;
    width: 33.3%;
    margin-bottom: 40px;
`;

export class Layout extends React.Component {
    componentDidMount () {
        this.props.fetchRequest()
    }

    componentWillReceiveProps(nextProps) {
    }

    render () {
        const { Fetch, showModal } = this.props;
        const Pending = Fetch.pending;

        return (
            <div>
                <Wrapper>
                    {
                        Pending?
                        <WrappedReactLoading>
                            <ReactLoading type="cylon" color="palevioletred"/>
                        </WrappedReactLoading>
                        :
                        Fetch.data.map((item, key) =>
                            <Item
                                onClick={() =>
                                    { showModal({
                                        modalType: "MODAL_STUDY",
                                        modalProps: item
                                    })}}
                                key={key}
                            >
                                {console.log(item)}
                                <Card
                                    item={item}
                                />
                            </Item>
                        )
                    }
                </Wrapper>
            </div>
        );
    }
};

Layout.propTypes = propTypes;
Layout.defaultTypes = defaultTypes;

export default Layout;
