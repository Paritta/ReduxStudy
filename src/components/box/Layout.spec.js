import React from "react";
import { Layout } from "./Layout";
import "jest-styled-components";
import ReactTestRenderer from "react-test-renderer";

it("Layout render correctly", () => {
    const tree = ReactTestRenderer.create(
        <Layout fetchRequest={() => {}} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
