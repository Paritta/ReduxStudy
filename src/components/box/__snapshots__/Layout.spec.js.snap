// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout render correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 100px;
}

<div>
  <div
    className="c0"
  />
</div>
`;
